### ISSUE TITLE
ParseLiveListWidget build items two times

### ISSUE THREAD

**Issue by kharitonovAL** [parse-community/Parse-SDK-Flutter#525]
Hi everybody!

I build a clean new flutter project to check if my suggestions were correct, and yes - `ParseLiveListWidget` build it's child twice.
My ParseServer database have 9 records of `UserRequest` class. I created a class named `UserRequest`:
```dart
class UserRequest extends ParseObject implements ParseCloneable {
  static const String _keyTableName = 'UserRequest';

  @override
  clone(Map<String, dynamic> map) => UserRequest.clone()..fromJson(map);

  UserRequest() : super(_keyTableName);
  UserRequest.clone() : this();

  int get requestNumber => get<int>(keyRequestNumber);
  set requestNumber(int requestNumber) => set<int>(keyRequestNumber, requestNumber);
}
```

My `main.dart` is below:
```dart
const String parseAppId = 'XXX';
const String parseServerUrl = 'https://XXX/parse/';
const String liveQueryUrl = 'ws://XXX:1337/live';


void main() async {
  CoreStore coreStore = await CoreStoreSharedPrefsImp.getInstance();
  await Parse().initialize(
    parseAppId, parseServerUrl,
    clientKey: 'XXX',
    autoSendSessionId: true,
    coreStore: coreStore,
    liveQueryUrl: liveQueryUrl,
    registeredSubClassMap: <String, ParseObjectConstructor>{
      'UserRequest': () => UserRequest(),
    },
  );

  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text(widget.title),
        ),
        body: SafeArea(
            child: Container(
          child: Center(
            child: RaisedButton(
                child: Text('enter'),
                color: Colors.blue,
                textColor: Colors.white,
                onPressed: () async {
                  ParseUser user = await login(email: 'XXX', password: 'XXX');
                  Navigator.push(context, MaterialPageRoute(builder: (context) => LiveWidget()));
                }),
          ),
        )));
  }

  Future<ParseUser> login({String email, String password}) async {
    ParseUser user = ParseUser(email, password, email);
    ParseResponse response = await user.login();

    if (response.success) {
      return user;
    } else {
      return null;
    }
  }
}
```

And my parse live widget is below:
```dart
class LiveWidget extends StatefulWidget {
  @override
  State<StatefulWidget> createState() => _LiveWidgetState();
}

class _LiveWidgetState extends State<LiveWidget> {
  var op = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: SafeArea(child: SingleChildScrollView(child: _buildBody(context))),
    );
  }

  Widget _buildBody(BuildContext context) {
    return ParseLiveListWidget<UserRequest>(
      query: getUserRequestQuery(),
      reverse: false,
      lazyLoading: false,
      shrinkWrap: true,
      listLoadingElement: Center(child: CircularProgressIndicator()),
      childBuilder: (context, ParseLiveListElementSnapshot<UserRequest> snapshot) {
        return snapshot.failed
            ? Text('snapshot failed')
            : snapshot.hasData
                ? _buildList(context, snapshot)
                : null;
      },
    );
  }

  Widget _buildList(BuildContext context, ParseLiveListElementSnapshot<UserRequest> snapshot) {
    print('_buildList ${op += 1}, ${snapshot.loadedData.requestNumber}');
    UserRequest request = snapshot.loadedData;
    return Card(child: ListTile(title: Text('${request.requestNumber}')));
  }

  QueryBuilder getUserRequestQuery() {
    final QueryBuilder query = QueryBuilder<UserRequest>(UserRequest());
    query.orderByDescending(UserRequest.keyRequestNumber);
    return query;
  }
}
```

And when I launch app, the console prints the following:
```
_buildList 1, 9
_buildList 2, 8
_buildList 3, 7
_buildList 4, 6
_buildList 5, 5
_buildList 6, 4
_buildList 7, 3
_buildList 8, 2
_buildList 9, 1
_buildList 10, 9
_buildList 11, 8
_buildList 12, 7
_buildList 13, 6
_buildList 14, 5
_buildList 15, 4
_buildList 16, 3
_buildList 17, 2
_buildList 18, 1
```

what I'm doing wrong?

Thanks!

**Comment by kharitonovAL** [comment]
interesting moment: if I'll turn `lazyLoading` to `true` (in `ParseLiveListWidget`), then it load data only once!

**Comment by kharitonovAL** [comment]
> interesting moment: if I'll turn `lazyLoading` to `true` (in `ParseLiveListWidget`), then it load data only once!

No, it loads data twice, but first cycle there are `null` items. Here is the method:
```dart
Widget _buildBody(BuildContext context) {
    return ParseLiveListWidget<UserRequest>(
      query: getUserRequestQuery(),
      reverse: false,
      // lazyLoading: false,
      shrinkWrap: true,
      duration: Duration(seconds: 1),
      listLoadingElement: Center(child: CircularProgressIndicator()),
      childBuilder: (context, ParseLiveListElementSnapshot<UserRequest> snapshot) {

        print('snapshot: ${snapshot.error}'); // prints out logs

        return snapshot.failed
            ? Text('snapshot failed')
            : snapshot.hasData
                ? _buildList(context, snapshot)
                : null;
      },
    );
  }
```

and the logs:
```
snapshot: null
snapshot: null
snapshot: null
snapshot: null
snapshot: null
snapshot: null
snapshot: null
snapshot: null
snapshot: null
snapshot: null
_buildList 1, 9
snapshot: null
_buildList 2, 8
snapshot: null
_buildList 3, 7
snapshot: null
_buildList 4, 6
snapshot: null
_buildList 5, 4
snapshot: null
_buildList 6, 5
snapshot: null
_buildList 7, 3
snapshot: null
_buildList 8, 2
snapshot: null
_buildList 9, 1
```

**Comment by RodrigoSMarques** [comment]
We are closing issues that have been open for a long time without activity. 
This will make it easier to organize things from now on.
If the problem persists, please open a new issue.
Thanks.
